rcp/org.testeditor.rcp4.views.tcltestrun.tests/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUiTest.xtend

```diff
@@ -0,0 +1,122 @@
+package org.testeditor.rcp4.views.tcltestrun
+
+import com.google.inject.Injector
+import org.eclipse.core.resources.IResource
+import org.eclipse.jface.viewers.IStructuredSelection
+import org.eclipse.xtext.naming.QualifiedName
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.runners.MockitoJUnitRunner
+import org.testeditor.rcp4.tcltestrun.TclInjectorProvider
+import org.testeditor.tcl.dsl.ui.testlaunch.LaunchShortcutUtil
+
+import static org.junit.Assert.*
+import static org.mockito.Matchers.*
+import static org.mockito.Mockito.*
+import org.eclipse.core.resources.IFolder
+import org.eclipse.core.resources.IFile
+import org.eclipse.jdt.core.IJavaElement
+
+@RunWith(MockitoJUnitRunner)
+class TclLauncherUiTest {
+
+	@Mock
+	TclInjectorProvider tclInjectorProvider
+
+	@Mock
+	IStructuredSelection selection
+
+	@Mock
+	LaunchShortcutUtil launchUtil
+
+	TclLauncherUi launcherUi
+
+	@Before
+	def void setup() {
+		// given
+		val injector = mock(Injector)
+		when(tclInjectorProvider.get).thenReturn(injector)
+		when(injector.getInstance(LaunchShortcutUtil)).thenReturn(launchUtil)
+		launcherUi = new TclLauncherUi(tclInjectorProvider)
+	}
+
+	@Test
+	def void testGetTestCaseListOnOneSelection() {
+		// given
+		val resource = mock(IResource)
+		when(selection.firstElement).thenReturn(resource)
+		when(resource.toString).thenReturn("myprj/mypackage/myTest.tcl");
```

Comment:
The fact that we need to mock `toString` here makes me wonder about the implementation. As far as I can see, the result of `toString` on `IResource` is not part of its API [(IResource).](http://help.eclipse.org/neon/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fcore%2Fresources%2FIResource.html) So we should use something like `getProjectRelativePath()` or `getRawLocation()` if possible.